common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java

```diff
@@ -488,6 +489,40 @@ protected SecretKey loadSecretKey(@NonNull final KeyType keyType) throws IOExcep
         throw new GeneralSecurityException(ErrorStrings.UNKNOWN_ERROR);
     }
 
+    /**
+     * If needed, get the key from inactive broker (if there is one).
+     *
+     * @return true if the migration happened and succeeded. false otherwise.
+     */
+    public boolean migrateEncryptionKeyIfNeeded(@NonNull final String callingPackageName) throws GeneralSecurityException, IOException {
+        if (!(COMPANY_PORTAL_APP_PACKAGE_NAME.equalsIgnoreCase(callingPackageName) ||
+                AZURE_AUTHENTICATOR_APP_PACKAGE_NAME.equalsIgnoreCase(callingPackageName))) {
+            // Caller is not broker. No need to migrate.
+            return false;
+        }
+
+        mEncryptionKey = getKey();
+        if (mEncryptionKey != null){
+            return false;
+        }
+
+        mEncryptionKey = getKeyFromInactiveBroker(callingPackageName);
+        return mEncryptionKey != null;
+    }
+
+
+    protected SecretKey getKeyFromInactiveBroker(@NonNull final String callingPackageName) {
```

Comment:
what do you think of renaming `callingPackageName` to `activeBrokerPackageName` ?